### PR TITLE
feat: sidebar subscription messages

### PR DESCRIPTION
**Pull request by alexnm**
Started implementing the end of trial experience (XTD-322) but realized there are a few other messages that are a few lines of code away so went ahead and added them all into one PR, so it covers all cases.

BONUS: While here, I split the RowItem components into separate files in the Sidebar folder and cleaned up some code.

The combinatorial logic might be useful in overmind and exposed to other screens, so let's keep an eye for that.

Case 1: Personal free account
Screenshot 2022-11-02 at 17 43 10

Case 2: Team free account - Admin (no prior subscription)
Screenshot 2022-11-02 at 17 44 52

Case 3: Team free account - Admin (cancelled or expired subscription)
Screenshot 2022-11-02 at 17 45 19

Case 4: Team free account - User (no CTA)
Screenshot 2022-11-02 at 17 45 39

Case 5: Trial active - Admin
Screenshot 2022-11-02 at 17 47 43

Case 6: Trial active - User (no CTA)
Screenshot 2022-11-02 at 17 48 32

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/7102/commits/30ae6817cba759cc5f3c2425aecd338c75c30fa1 is at https://pr7102.build.csb.dev/s/new.

File: packages/app/src/app/utils/dateTime.test.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { getDaysUntil } from './dateTime';
+
+describe(getDaysUntil, () => {
```
Comment (tristandubbeld) [inline packages/app/src/app/utils/dateTime.test.ts:3]: 🙌

File: packages/app/src/app/utils/dateTime.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export const getDaysUntil = (target: Date): number | null => {
+  if (!target) {
+    return null;
+  }
```
Comment (tristandubbeld) [inline packages/app/src/app/utils/dateTime.ts:4]: Very safe haha :)

**Review by tristandubbeld — APPROVED** [review]
Looks good! Also, great PR description with all the screenshots.

File: packages/app/src/app/pages/Dashboard/Sidebar/BottomMessages/TrialExpiring.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+      {daysLeft > 1 && <>{daysLeft} days left on your trial.</>}
+    </Text>
+    <Text css={{ color: '#999', fontWeight: 400, fontSize: 12 }}>
+      After this period, your Team Pro subscription will be automatically
+      renewed.
+    </Text>
+    {isAdmin && (
+      <Link
+        as={RouterLink}
+        to="/dashboard/settings"
```
Comment (olarclara) [inline packages/app/src/app/pages/Dashboard/Sidebar/BottomMessages/TrialExpiring.tsx:22]: Should this link to this screen?

Screen Shot 2022-10-30 at 22 28 36

**Review by olarclara — APPROVED** [review]

**Review by tristandubbeld — APPROVED** [review]
Checked the latest commit, still looks good to me!

**Pull request merged**